### PR TITLE
#371 - ParsedCatalog added

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #371 
`ParsedCatalog` added

### CODE REVIEW

File: src/test/java/com/artipie/docker/misc/ParsedCatalogTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * @since 0.10
+ */
+class ParsedCatalogTest {
+
+    @Test
+    void parsesNames() {
+        MatcherAssert.assertThat(
+            new ParsedCatalog(
+                () -> new Content.From("{\"repositories\":[\"one\",\"two\"]}".getBytes())
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/docker/misc/ParsedCatalogTest.java:44]: @olegmoz I'd suggest to add test for the case when origin catalog is empty or contains empty `{}` json

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz please check suggestion

**Comment by olegmoz** [comment]
@olenagerasimova thanks, added more tests

**Review by olenagerasimova — APPROVED** [review]
@olegmoz good to merge
